packages/trader/src/Modules/Trading/Containers/trade-chart.tsx

```diff
@@ -4,22 +4,29 @@ import { isDesktop } from '@deriv/shared';
 import { observer, useStore } from '@deriv/stores';
 import { useTraderStore } from 'Stores/useTraderStores';
 import { ChartBottomWidgets } from './chart-widgets';
-import SmartChartSwitcher from './smart-chart-switcher';
 import AccumulatorsChartElements from '../../SmartChart/Components/Markers/accumulators-chart-elements';
 import ToolbarWidgets from '../../SmartChart/Components/toolbar-widgets';
-import ToolbarWidgetsBeta from '../../SmartChartBeta/Components/toolbar-widgets';
-import AllMarkers from '../../SmartChart/Components/all-markers.jsx';
 import type { TBottomWidgetsParams } from './trade';
+import { SmartChart } from 'Modules/SmartChart';
+import { ChartBarrierStore } from 'Stores/Modules/SmartChart/chart-barrier-store';
 
 type TTradeChartProps = {
     bottomWidgets?: (props: TBottomWidgetsParams) => React.ReactElement;
     has_barrier?: boolean;
     is_accumulator: boolean;
-    topWidgets: () => React.ReactElement;
+    charts_ref: React.MutableRefObject<HTMLDivElement>;
+    topWidgets: (() => JSX.Element) | null | undefined;
+    children?: React.ReactNode;
 };
+//TODO: will add props later
+const SmartChartWithRef = React.forwardRef((props: any, ref: React.ForwardedRef<HTMLElement | null>) => (
```

Comment:
<SmartChart doesn't use innerRef, and it's also not a class component which could use ref without assigning it to a node. We can get rid of SmartChartWithRef like it's done on master.

Reply:
sure , thankyou